cmd/ncproxy/server_test.go

```diff
@@ -0,0 +1,1094 @@
+
+	type config struct {
+		name          string
+		containerID   string
+		nicID         string
+		endpointName  string
+		errorExpected bool
+	}
+	tests := []config{
+		{
+			name:          "DeleteNIC returns no error",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: false,
+		},
+		{
+			name:          "DeleteNIC returns error with blank container ID",
+			containerID:   "",
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "DeleteNIC returns error with blank nic ID",
+			containerID:   containerID,
+			nicID:         "",
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "DeleteNIC returns error with blank endpoint name",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  "",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(_ *testing.T) {
+			req := &ncproxygrpc.DeleteNICRequest{
+				ContainerID:  test.containerID,
+				NicID:        test.nicID,
+				EndpointName: test.endpointName,
+			}
+
+			_, err := gService.DeleteNIC(ctx, req)
+			if test.errorExpected && err == nil {
+				t.Fatalf("expected DeleteNIC to return an error")
+			}
+			if !test.errorExpected && err != nil {
+				t.Fatalf("expected DeleteNIC to return no error, instead got %v", err)
+			}
+		})
+	}
+}
+
+func TestModifyNIC(t *testing.T) {
+	ctx := context.Background()
+
+	// setup test ncproxy grpc service
+	agentCache := newComputeAgentCache()
+	gService := newGRPCService(agentCache)
+
+	var (
+		containerID = "test-container"
+		testNICID   = "test-nic-id"
+	)
+
+	// create mock compute agent service
+	computeAgentCtrl := gomock.NewController(t)
+	mockedService := ncproxyMock.NewMockComputeAgentService(computeAgentCtrl)
+
+	// populate agent cache with mocked service for test
+	agentCache.put(containerID, mockedService)
+
+	// setup expected mocked calls
+	mockedService.EXPECT().ModifyNIC(gomock.Any(), gomock.Any()).Return(&computeagent.ModifyNICInternalResponse{}, nil).AnyTimes()
+
+	// create test network
+	network, err := createTestNATNetwork()
+	if err != nil {
+		t.Fatalf("failed to create test network with %v", err)
+	}
+	defer func() {
+		_ = network.Delete()
+	}()
+
+	// create test endpoint
+	endpointName := "test-endpoint-name"
+	endpoint, err := createTestEndpoint(endpointName, network.Id)
+	if err != nil {
+		t.Fatalf("failed to create test endpoint with %v", err)
+	}
+	defer func() {
+		_ = endpoint.Delete()
+	}()
+
+	iovOffloadOff := &ncproxygrpc.IovEndpointPolicySetting{
+		IovOffloadWeight: 0,
+	}
+
+	type config struct {
+		name              string
+		containerID       string
+		nicID             string
+		endpointName      string
+		iovPolicySettings *ncproxygrpc.IovEndpointPolicySetting
```

Comment:
CI's going to fail on this as it's ws2019, the policy wasn't present.

Reply:
I have a block on the test for 21h1

Reply:
Huzzah, sweet. Thank you!